resources/elasticbeanstalk_configuration_options.go

```diff
@@ -0,0 +1,163 @@
+				Type:        schema.TypeInt,
+			},
+			{
+				Name:        "name",
+				Description: "The name of the configuration option.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "namespace",
+				Description: "A unique namespace identifying the option's associated AWS resource.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "regex_label",
+				Description: "A unique name representing this regular expression.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Regex.Label"),
+			},
+			{
+				Name:        "regex_pattern",
+				Description: "The regular expression pattern that a string configuration option value with this restriction must match.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Regex.Pattern"),
+			},
+			{
+				Name:        "user_defined",
+				Description: "An indication of whether the user defined this configuration option:  * true : This configuration option was defined by the user",
+				Type:        schema.TypeBool,
+			},
+			{
+				Name:        "value_options",
+				Description: "If specified, values for the configuration option are selected from this list.",
+				Type:        schema.TypeStringArray,
+			},
+			{
+				Name:        "value_type",
+				Description: "An indication of which type of values this option has and whether it is allowable to select one or more than one of the possible values:  * Scalar : Values for this option are a single selection from the possible values, or an unformatted string, or numeric value governed by the MIN/MAX/Regex constraints.  * List : Values for this option are multiple selections from the possible values.  * Boolean : Values for this option are either true or false .  * Json : Values for this option are a JSON representation of a ConfigDocument.",
+				Type:        schema.TypeString,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchElasticbeanstalkConfigurationOptions(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+
+	c := meta.(*client.Client)
+	svc := c.Services().ElasticBeanstalk
+	var config elasticbeanstalk.DescribeEnvironmentsInput
+	allEnvs := make([]types.EnvironmentDescription, 0)
+	for {
+		response, err := svc.DescribeEnvironments(ctx, &config, func(options *elasticbeanstalk.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return err
+		}
+		allEnvs = append(allEnvs, response.Environments...)
+		if aws.ToString(response.NextToken) == "" {
+			break
+		}
+		config.NextToken = response.NextToken
+	}
+
+	var configOptionsIn elasticbeanstalk.DescribeConfigurationOptionsInput
```

Comment:
nit: I'd move this variable inside the `for` loop to make sure it doesn't carry any state/data over different requests.